index.js

```diff
@@ -37,6 +37,23 @@ const pDebounce = (fn, wait, options = {}) => {
 	};
 };
 
+pDebounce.promise = function_ => {
+	let currentPromise;
+
+	return async (...arguments_) => {
```

Comment:
I think that async functions aren't supported in Node 6 https://node.green/#ES2017-features-async-functions-return

Reply:
Should I migrate to Node.js 10 in this PR?

Reply:
I think the alternative is to just use .then instead of the `try` block, maybe:

```js
return currentPromise.then(result => {
 	currentPromise = undefined;
	return result;
}, error => {
 	currentPromise = undefined;
	throw result;
});
```

The Node 12+ breaking wave is coming in just 4 months: https://github.com/sindresorhus/meta/discussions/15

Reply:
This is not a commonly used package, so I'm happy to just require Node.js 12 now, if what you need this for can use Node.js 12 too?